### PR TITLE
ref(ts): convert plugin config to TS

### PR DESCRIPTION
**Pull request by scefali**

### CODE REVIEW

**Review by leedongwei — COMMENTED** [review]
Should `Props['enabled']` be optional? Seems to have an error in `components/pluginList.tsx`.

**Comment by scefali** [comment]
> Should `Props['enabled']` be optional? Seems to have an error in `components/pluginList.tsx`.

@leedongwei You are right. Fixed.

**Review by leedongwei — APPROVED** [review]
LGTM. Before merging, could you change Line 118 to use `??`, I can't put a comment on it because it's outside of the diff.

File: src/sentry/static/sentry/app/components/pluginConfig.tsx [hunk 1]
```diff
@@ -16,9 +16,25 @@ import PluginIcon from 'app/plugins/components/pluginIcon';
...
+import {Organization, Project, Plugin} from 'app/types';
+import {Client} from 'app/api';
+
+type Props = {
+  organization: Organization;
+  project: Project;
+  data: Plugin;
+  onDisablePlugin: (data: Plugin) => void;
+  enabled?: boolean;
+  api: Client;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/pluginConfig.tsx:28]:
```suggestion
  api: Client;
   enabled?: boolean;
```

File: src/sentry/static/sentry/app/components/pluginConfig.tsx [hunk 2]
```diff
@@ -16,9 +16,25 @@ import PluginIcon from 'app/plugins/components/pluginIcon';
...
+};
+
+type State = {
+  loading?: boolean;
+  testResults: string;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/pluginConfig.tsx:33]:
```suggestion
  testResults: string;
   loading?: boolean;
```

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/pluginConfig.tsx:33]: we usually declare optional props at the end of the object 😉

**Comment by scefali** [comment]
@leedongwei 
>LGTM. Before merging, could you change Line 118 to use ??, I can't put a comment on it because it's outside of the diff.

Unfortunately, I don't think I can because we need to treat `false` differently than `undefined` for `this.props.enabled`.

**Review by k-fish — APPROVED** [review]

**Pull request merged**